# E-commerce Data Analysis Project

## 🎯 Project Objectives

The goal of this project is to explore and analyze customer order data to extract meaningful business insights and build interactive dashboards. Specifically, the project aims to:

1. **Customer Behavior Analysis**  
   Understand who spends more, identify high-frequency buyers, assess age and gender trends, and evaluate the impact of shipping fees on customer purchasing behavior.

2. **Sales and Revenue Trend Tracking**  
   Analyze revenue and order trends over time (monthly, quarterly), identify seasonal peaks, track performance by region, product, and customer demographics.

3. **Product Performance Insights**  
   Evaluate which products are top performers based on revenue and quantity sold, assess delivery success rates, identify region-specific vs. universally popular products.

4. **Business KPI Monitoring**  
   Track key performance indicators such as average order value (AOV), order volume, units sold, shipping status completion rates, and customer repeat rate.

5. **Dashboard Development**  
   Create interactive and visually engaging dashboards in Power BI for stakeholder communication and ongoing business monitoring.

## Load Data

In [1]:
# Import libraries
import pandas as pd

In [11]:
# Load data
df = pd.read_csv("../data/raw_orders.csv")
df.shape
df.info()
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 12 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Customer ID      1000 non-null   object 
 1   Gender           1000 non-null   object 
 2   Region           950 non-null    object 
 3   Age              900 non-null    float64
 4   Product Name     1000 non-null   object 
 5   Category         1000 non-null   object 
 6   Unit Price       1000 non-null   float64
 7   Quantity         1000 non-null   int64  
 8   Total Price      1000 non-null   int64  
 9   Shipping Fee     1000 non-null   float64
 10  Shipping Status  950 non-null    object 
 11  Order Date       1000 non-null   object 
dtypes: float64(3), int64(2), object(7)
memory usage: 93.9+ KB


Customer ID  Gender Region   Age Product Name     Category  Unit Price  \
0    CUST0268    Male  North   NaN      Monitor  Electronics       300.0   
1    CUST0046    Male   West  22.0   Headphones  Accessories       100.0   
2    CUST0169  Female  South  54.0      Monitor  Electronics       300.0   
3    CUST0002    Male  North  23.0   Headphones  Accessories       100.0   
4    CUST0173  Female  South   NaN       Laptop  Electronics      1500.0   

   Quantity  Total Price  Shipping Fee Shipping Status  Order Date  
0         5         1500         13.31        Returned  2023-12-08  
1         2          200          6.93      In Transit  2023-04-09  
2         1          300         11.31        Returned  2023-08-28  
3         5          500         12.22       Delivered  2023-01-18  
4         3         4500          5.40       Delivered  2023-01-19

In [8]:
# Check for null values
# df.isnull().sum()

In [15]:
# Drop null values
df.dropna(inplace=True)
# Recheck shape
len(df)

810

In [16]:
# Type conversions
# Convert `Age` column to `int`
df["Age"] = df["Age"].astype(int)
# Total price to float
df["Total Price"] = df["Total Price"].astype(float)
# Order Date to date
df["Order Date"] = pd.to_datetime(df["Order Date"])

In [18]:
# Confirm type conversion
# df.info()

## Initial EDA

## Data Cleaning

## Feature Engineering

## Saving Results

## Summary of Cleaning Steps

## Notes

1. Removed all null values - leading to a loss of about 19% of the data